crates/flux-fhir-analysis/src/conv/mod.rs

```diff
@@ -597,11 +598,20 @@ impl<'genv, 'tcx: 'genv, P: ConvPhase<'genv, 'tcx>> ConvCtxt<P> {
                     })
                     .collect(),
             );
+
+            let requires = adt_def
+                .invariants()
+                .iter()
+                .map(|inv| inv.apply(&idx))
+                .collect();
+            let requires = List::from_vec(requires);
```

Comment:
I think you can collect directly into a `List`, haven't check if it works though.

```suggestion
            let requires = adt_def
                .invariants()
                .iter()
                .map(|inv| inv.apply(&idx))
                .collect();
```

Reply:
indeed looks like that works, thanks!